frontend/src/routes/events/Events.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useState } from 'react'
+import NumberPagination from '../../components/NumberPagination/NumberPagination'
+import { getFormattedPeriod, getSummary } from '../../util/functions'
+import { useGeneralContext } from '../../contexts/generalState'
+import { getPaginatedEvents } from '../../api/event'
+import { EventType } from '../../data/event'
+import ScrollToTop from '../../components/ScrollToTop/ScrollToTop'
+import { useNavigate } from 'react-router-dom'
+
+import './Events.scss'
+
+interface EventState {
+  events: EventType[]
+  loading: boolean
+}
+
+function Events() {
+  const [eventRequest, setEventRequest] = useState<EventState>({ events: [], loading: true })
+  const [page, setPage] = useState<number>(1)
+  const [totalPages, setTotalPages] = useState<number>(1)
+  const { backgroundImages } = useGeneralContext()
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    ;(async () => {
+      window.scroll({ top: 0, behavior: 'smooth' })
+      setEventRequest({ events: [], loading: true })
+      getPaginatedEvents(page).then((response) => {
+        setEventRequest({ events: response.events, loading: false })
+        setTotalPages(response.pageCount)
+      })
+    })()
+  }, [page])
+
+  const goToEventPage = (id: number) => {
+    navigate(`/evenimente/${id}`)
+  }
+
+  const { loading, events } = eventRequest
```

Comment:
What happens if the loading result never comes back because of an error. Can a timeout and an elegant fail safe be added? I guess this would be applicable to all places where a react component is waiting for an api response.

Reply:
we can think of something and add it later for all pages